src/rest-server/src/models/v2/job.js

```diff
@@ -232,9 +233,13 @@ const prepareContainerScripts = async (frameworkName, userName, config, rawConfi
     JSON.stringify(frameworkDescription, null, 2), userName, '644')
   );
   // upload original config file to hdfs
+  // mask secrets field before uploading
+  let maskRawConfig = rawConfig + '\nZ';
```

Comment:
Regex in JavaScript does not support perl regex `\Z`, so add a newline at the end manually.

Reply:
ut please.

---
In reply to: [283308207](https://github.com/microsoft/pai/pull/2752#discussion_r283308207) [](ancestors = 283308207)

Reply:
split into a separate function and added unit test